dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/ror/RorIndexingService.java

```diff
@@ -31,8 +34,24 @@
     private RorConverter rorConverter;
 
     @Asynchronous
-    public Future<UpdateResponse> indexRorRecordAsync(RorData rorData) {
-        return CompletableFuture.supplyAsync(() -> indexRorRecord(rorConverter.toSolrDto(rorData)));
+    public Future<UpdateResponse> indexRorRecordsAsync(Collection<RorData> rorData) {
+
+        List<RorDto> convertedData = rorData.stream()
+                                            .map(ror -> rorConverter.toSolrDto(ror))
+                                            .collect(Collectors.toList());
+
+        String failedRorIds = rorData.stream()
+                                     .map(RorData::getRorId)
+                                     .collect(Collectors.joining(","));
+
+        CompletableFuture<UpdateResponse> updateResult = CompletableFuture.supplyAsync(() -> {
+            Try.of(() -> solrServer.addBeans(convertedData))
+               .onFailure(throwable -> logger.log(Level.WARNING, "Unable to add ror records with ror ids: " + failedRorIds, throwable));
```

Comment:
Is it wise to log 100 000 ids via logger?

Reply:
In theory you can upload different dump sizes and there has to be a way to identify which data has failed, then maybe we should just log first id from collection?

Reply:
Maybe we could include count of records and then let's say ids of 10 first ids?